### PR TITLE
OMHIREDIS:: ADD 'SET' MODE

### PR DESCRIPTION
**Pull request by frikilax**
- new mode 'set' to send SET/SETEX commands
- new parameter 'expiration' to send SETEX instead of SET commands (only applicable to 'set' mode)
- fixes to missing frees

### CODE REVIEW

**Comment by rgerhards** [comment]
assigning milestone tentatively

File: contrib/omhiredis/omhiredis.c [hunk 1]
```diff
@@ -213,6 +238,8 @@ static rsRetVal writeHiredis(uchar* key, uchar *message, wrkrInstanceData_t *pWr
 	}
 
 finalize_it:
+	if (formattedMsg)
```
Comment (rgerhards) [inline contrib/omhiredis/omhiredis.c]: This check should be avoided. `free(NULL)` is well-defined.

File: contrib/omhiredis/omhiredis.c [hunk 2]
```diff
@@ -185,7 +194,9 @@ static rsRetVal writeHiredis(uchar* key, uchar *message, wrkrInstanceData_t *pWr
 	 * happened, in which case abort. otherwise
 	 * increase our current pipeline count
 	 * by 1 and continue. */
-	int rc;
+	int rc, expire;
+	size_t msgLen;
+	char *formattedMsg = NULL;
```
Comment (rgerhards) [inline contrib/omhiredis/omhiredis.c]: The compiler is right: if this line is not reached, the variable exists, is used during cleanup, but is undefined.

Suggestion: move up to top of function.

File: contrib/omhiredis/omhiredis.c [hunk 3]
```diff
@@ -121,6 +124,12 @@ CODESTARTfreeInstance
 	if (pData->server != NULL) {
 		free(pData->server);
 	}
+	if (pData->key)
```
Comment (rgerhards) [inline contrib/omhiredis/omhiredis.c]: remove redundant NULL checks

**Comment by rgerhards** [comment]
please add doc to https://github.com/rsyslog/rsyslog-doc/

**Comment by frikilax** [comment]
> please add doc to https://github.com/rsyslog/rsyslog-doc/

Done in rsyslog/rsyslog-doc#876

**Pull request merged**